### DISCUSSION TITLE
[Feature]: Ability to show/hide any notification control

### DISCUSSION THREAD

**Discussion by amitkumarcoding** [TheWidlarzGroup/react-native-video#4938] [Ideas]
### What problem are you trying to solve?

Want an option to hide or show specific controls in a React Native video music (notification) control, as in some cases you may not want a particular control to be visible.

### Proposed solution

An option to show or hide any notification control individually.

**Comment by freeboub** [comment]
@amitmehtacode If you want to enhance the behavior, you should improve [controlsStyles](https://thewidlarzgroup.github.io/react-native-video/component/props#controlsstyles) property

**Comment by amitkumarcoding** [comment]
@freeboub 
But how can I hide forward and backward icon on version 6.0.0
as controlsStyles are not supported in 6.0.0 version

**Comment by freeboub** [comment]
no

**Comment by amitkumarcoding** [comment]
@freeboub 
also checked on version 6.1.1 after adding
```
controlsStyles={{
          hideSeekBar: true
        }}
```

The seekBar is still not hidden.

**Comment by freeboub** [comment]
> @freeboub also checked on version 6.1.1 after adding
> 
> ```
> controlsStyles={{
>           hideSeekBar: true
>         }}
> ```
> 
> The seekBar is still not hidden.

Just retested in sample app, it works fine 🤔

**Comment by amitkumarcoding** [comment]
@freeboub please find the recording 
version: 6.1.1

https://github.com/TheWidlarzGroup/react-native-video/assets/90309641/8937d9af-418a-4050-a875-83ce723a8c87

**Comment by amitkumarcoding** [comment]
@freeboub 
how can I remove the forward and backward icons?
I only want the play and pause icons in it.

**Comment by YangJonghun** [comment]
@amitmehtacode 
The showNotificationControls prop enables the Android media session. I'm not sure why you want to disable the backward and forward buttons, but if you want to block functionality, you should remove the listener rather than blocking the UI, because the media session controls all external controls together, including the physical buttons, not just notifications.

**Comment by amitkumarcoding** [comment]
@YangJonghun 
In one of my use case I only want to show play and pause icon. Also I highly appreciated if you can guide me for the same with code how do I remove the listener. Also if it is possible to hide the icon ?

**Comment by YangJonghun** [comment]
@amitmehtacode 
https://github.com/TheWidlarzGroup/react-native-video/blob/master/android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackService.kt
https://github.com/TheWidlarzGroup/react-native-video/blob/master/android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackCallback.kt

you need to patch above codes for build media session without predefined buttons(and callbacks)

**Comment by seyedmostafahasani** [comment]
> > @freeboub also checked on version 6.1.1 after adding
> > ```
> > controlsStyles={{
> >           hideSeekBar: true
> >         }}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > The seekBar is still not hidden.
> 
> Just retested in sample app, it works fine 🤔

@freeboub hideSeekBar just works on video play controller. I think he wants to hide it on notification.

**Comment by amitkumarcoding** [comment]
> > > @freeboub also checked on version 6.1.1 after adding
> > > ```
> > > controlsStyles={{
> > >           hideSeekBar: true
> > >         }}
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > The seekBar is still not hidden.
> > 
> > 
> > Just retested in sample app, it works fine 🤔
> 
> @freeboub hideSeekBar just works on video play controller. I think he wants to hide it on notification.

Yes

**Comment by amitkumarcoding** [comment]
@freeboub I want to hide these 3 controls from the notification screen. How can this be done? Additionally, there should be an option to show or remove them through props.

![Screenshot_20240528_222257_Chrome](https://github.com/TheWidlarzGroup/react-native-video/assets/90309641/3a895fe3-caae-41cb-b5ef-1e6eae3194c9)

**Comment by KrzysztofMoch** [comment]
For now there is not prop for doing it - if you need it for commercial project you can contact TheWidlarzGroup [here](https://www.thewidlarzgroup.com/)

**Comment by paulrinaldi** [comment]
@amitmehtacode https://github.com/TheWidlarzGroup/react-native-video/blob/0a1085ce03152d58d98da408dbe79e76fa5ebc1a/android/src/main/java/com/brentvatne/exoplayer/VideoPlaybackService.kt#L123 This is where those play and pause icons and functionality are created. This is where you can make some edits and then run the `examples/basic` with `yarn run android`.

**Comment by amayanga** [comment]
@YangJonghun @paulrinaldi would the seekbar also be in VideoPlaybackService.kt? I want to hide everything other than pause/play.

**Comment by paulrinaldi** [comment]
I hid everything by setting controls to false

**Comment by amayanga** [comment]
I don't want to hide everything. I still want to show play/pause on the notification controls.

**Comment by TWG-AI** [comment]
In v7, lockscreen/notification controls are toggled via `player.showNotificationControls = true/false`. You can show or hide specific controls on a per-player basis.

> I'm an AI — verify platform-specific behavior in the v7 docs.
